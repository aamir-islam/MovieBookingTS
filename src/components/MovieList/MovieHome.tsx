import React,{useState} from 'react'
import MovieList from './MovieList'
import { ListHeadingStyle,HeaderStyle,InputBoxStyle } from '../Styles/Card.style';

type Props = {

}

const MovieHome = (props: Props) => {
    const [input, setInput] = useState<string>("")
  return (
    <HeaderStyle>
          <ListHeadingStyle>Book Tickets</ListHeadingStyle>
          <InputBoxStyle type="text"
           placeholder="Search" 
           onChange={(e:React.ChangeEvent<HTMLInputElement>) => setInput(e.target.value)}
          />
          <MovieList input={input}/>
    </HeaderStyle>
  )
}

export default MovieHome