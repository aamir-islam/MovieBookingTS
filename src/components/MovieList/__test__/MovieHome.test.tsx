import React from 'react';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import MovieHome from '../MovieHome'

afterEach(cleanup);
describe("test for movie card", () => {
    test('should take a snapshot', () => {
        const { container } = render(<MovieHome />);
        expect(container).toMatchSnapshot()
    })


    test("Check input value", () => {
        render(<MovieHome />)
        const inputValue = screen.getByPlaceholderText<HTMLInputElement>(/Search/i)
        expect(inputValue).toBeInTheDocument()
        fireEvent.change(inputValue, { target: { value: "batman" } })
        expect(inputValue.value).toBe("batman");
    })



})


