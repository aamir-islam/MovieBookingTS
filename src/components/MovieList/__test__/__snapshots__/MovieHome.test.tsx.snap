// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for movie card should take a snapshot 1`] = `
<div>
  <header
    class="sc-gsnTZi ToOMS"
  >
    <h1
      class="sc-dkzDqf laljGH"
    >
      Book Tickets
    </h1>
    <input
      class="sc-hKMtZM jdtSxD"
      placeholder="Search"
      type="text"
    />
    <article
      class="sc-jSMfEi jQBanV"
    >
      <article
        class="sc-papXJ dPuUIy"
      >
        <h2
          class="sc-jqUVSM fWYeLJ"
        >
          Sorry,there is no result for keyword you searched
        </h2>
        <img
          alt="pnf"
          class="sc-kDDrLX kdEeEq"
          src="pageNotFound.png"
        />
      </article>
    </article>
  </header>
</div>
`;
